lib/web_ui/lib/src/engine/semantics/focusable.dart

```diff
@@ -164,6 +189,7 @@ class AccessibilityFocusManager {
     }
 
     target.element.removeEventListener('focus', target.domFocusListener);
+    target.element.removeEventListener('blur', target.domBlurListener);
```

Comment:
is it possible to tell the difference between framework request focus and user interacting dom focus by the Event object in the listener?

Reply:
I'm not aware of one for focus/blur events specifically. The closest thing available is the [isTrusted property](https://developer.mozilla.org/en-US/docs/Web/API/Event/isTrusted), but it only detects `dispatchEvent`, which is not enough. Clicks can be differentiated by looking at their `screenX/Y` properties ([source](https://cloudfour.com/thinks/detecting-if-an-event-was-triggered-by-a-user-or-by-javascript/)), but focus/blur doesn't have coordinates.